### PR TITLE
Tidy up chairman tests

### PR DESCRIPTION
**Pull request by newhoggy**
Tidy up chairman tests support code.  Assert that chairman exits successfully.

* Replace `evalM . liftIO` with `evalIO`
* Add function documentation for many functions
* Move `argQuote` quote into its own module because it didn't fit in with the rest of the module
* Move `threadDelay` to `Concurrent` module.
* Modify function signatures to abstract type to maximise re-use.
* Modify `createFileLink` to create a symbolic link on posix systems.
* Simplify type for `waitForProcess` and `waitSecondsForProcess` to not use `Maybe`

### CODE REVIEW

File: cardano-node-chairman/test/Test/Cardano/Node/Chairman/ByronShelley.hs [hunk 1]
```diff
@@ -648,10 +649,13 @@ prop_chairman = H.propertyOnce . H.workspace "chairman" $ \tempAbsPath -> do
         )
       )
 
-    void $ H.waitSecondsForProcess 110 hProcess
+    chairmanResult <- H.waitSecondsForProcess 110 hProcess
```
Comment (erikd) [inline cardano-node-chairman/test/Test/Cardano/Node/Chairman/ByronShelley.hs:652]: :100:

File: cardano-node-chairman/src/Chairman/Hedgehog/Base.hs [hunk 2]
```diff
@@ -124,112 +121,131 @@ workspace prefixPath f = GHC.withFrozenCallStack $ do
 --
 -- The directory will be deleted if the block succeeds, but left behind if
 -- the block fails.
-moduleWorkspace :: HasCallStack => FilePath -> (FilePath -> Integration ()) -> Integration ()
+moduleWorkspace :: (MonadTest m, MonadIO m, HasCallStack) => FilePath -> (FilePath -> m ()) -> m ()
```
Comment (erikd) [inline cardano-node-chairman/src/Chairman/Hedgehog/Base.hs:124]: This makes sense IMO.

**Review by erikd — APPROVED** [review]
LGTM!

**Comment by newhoggy** [comment]
bors merge

**Comment by newhoggy** [comment]
bors merge

**Pull request merged**